ares/n64/cpu/interpreter-ipu.cpp

```diff
@@ -118,8 +118,9 @@ auto CPU::BREAK() -> void {
 }
 
 auto CPU::CACHE(u8 operation, cr64& rs, s16 imm) -> void {
-  auto access = devirtualize(rs.u64 + imm);
+  auto access = devirtualize<Read>(rs.u64 + imm);
   if (!access) return;
```

Comment:
This can skip the vaddrAlignedError check, which seems like a change in behavior. Is this correct?

Similar questions apply to LL and LLD.

Reply:
CPU::CACHE calls vaddrAlignedError explicitly just after.
LL/LLD calls CPU::read that also calls vaddrAlignedError.

Reply:
Yes, but it can be skipped by the early return in CACHE, and the read call in LL/LLD is also inside a conditional. Before this PR, the alignment check ran at the top of devirtualize() before anything else.

Reply:
Thanks, this is indeed a bug not covered by n64-systemtest. I've verified on hardware that address errors come always before TLB errors (and [added a test](https://github.com/lemmy-64/n64-systemtest/pull/80)).

To fix this, I've had to add a new template parameter to multiplex `devirtualize()` also by access size (as aligned errors depend on the access size). I've also had to reinstate the hack for which CPU writes can request not to emit align errors (used by SWR -- and should be used by SDR -- while we don't properly support masked writes in the SysAD bus implementation).

Reply:
Thanks for digging into it!